inline-c-cpp/inline-c-cpp.cabal

```diff
@@ -32,16 +32,20 @@ library
                      , containers
   hs-source-dirs:      src
   default-language:    Haskell2010
-  ghc-options:         -Wall -optc-xc++ -optc-std=c++11
+  ghc-options:         -Wall
   include-dirs:        include
   install-includes:    HaskellException.hxx HaskellStablePtr.hxx
   extra-libraries:     stdc++
-  c-sources:           cxx-src/HaskellException.cxx cxx-src/HaskellStablePtr.cxx
-  cc-options:          -Wall -std=c++11
+  cxx-sources:         cxx-src/HaskellException.cxx cxx-src/HaskellStablePtr.cxx
+  -- cxx-options is for .cxx _file_ compilation whereas -optcxx is
+  -- for TemplateHaskell, so we duplicate the options
+  cxx-options:         -Wall -std=c++11
+  if impl(ghc >= 8.10)
```

Comment:
Could you explain why it is OK to only set these options in GHC >= 8.10? I suspect that this is related to your comment

> This also means that we don't set c++11 for inline-c-cpp itself. This should be ok, because

As in, we don't do that for GHC < 8.10. In any case, please explain the tradeoffs in a comment here.